### PR TITLE
Add extract() function

### PR DESCRIPTION
**Pull request by bwiernik**
Similar to `$` or `dplyr::pull()` with added ability to set extracted vector names to be the rownames of the data frame

### CODE REVIEW

**Comment by strengejacke** [comment]
I think in line with the easystats API, `select` should accept a character vector. @DominiqueMakowski ?

**Comment by bwiernik** [comment]
It does accept a character value:

```
#'   - a literal variable name (e.g., `column_name`)
#'   - a character vector with the variable name (e.g., `"column_name"`)
#'   - a positive integer, giving the position counting from the left
#'   - a negative integer, giving the position counting from the right.
```

**Comment by strengejacke** [comment]
Ah, nice! Just one minor thing, add `stats::`  to `setNames`.

File: R/data_extract.R [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  name <- eval(substitute(name), nl, parent.frame())
+  if (is.numeric(name) && name < 0) {
+    name <- length(data) + name + 1
+  } else if (is.numeric(name) && name == 0) {
+    name <- rownames(data)
+  } else if (is.character(name) && name == "row.names") {
+    name <- rownames(data)
+  }
+  if (length(name) == 1) {
+    setNames(data[,select, drop = TRUE], data[,name, drop = TRUE])
```
Comment (bwiernik) [inline R/data_extract.R:61]:
```suggestion
    stats::setNames(data[,select, drop = TRUE], data[,name, drop = TRUE])
```

File: R/data_extract.R [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  } else {
+    setNames(data[,select, drop = TRUE], name)
```
Comment (bwiernik) [inline R/data_extract.R:63]:
```suggestion
    stats::setNames(data[,select, drop = TRUE], name)
```

File: R/data_extract.R [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    name <- length(data) + name + 1
+  } else if (is.numeric(name) && name == 0) {
+    name <- rownames(data)
+  } else if (is.character(name) && name == "row.names") {
+    name <- rownames(data)
+  }
+  if (length(name) == 1) {
+    stats::setNames(data[,select, drop = TRUE], data[,name, drop = TRUE])
+  } else {
+    setNames(data[,select, drop = TRUE], name)
```
Comment (bwiernik) [inline R/data_extract.R:63]:
```suggestion
    stats::setNames(data[,select, drop = TRUE], name)
```

**Pull request merged**